subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheToolingApiInvocationIntegrationTest.groovy

```diff
@@ -99,6 +102,74 @@ class ConfigurationCacheToolingApiInvocationIntegrationTest extends AbstractConf
         outputContains("Reusing configuration cache.")
     }
 
+    def "configuration cache is disabled for model building actions"() {
+
+        given:
+        file("buildSrc/src/main/groovy/my/My.groovy") << """
+            package my
+
+            import org.gradle.tooling.provider.model.ToolingModelBuilder
+            import org.gradle.api.Project
+
+            class MyModelBuilder implements ToolingModelBuilder {
+                boolean canBuild(String modelName) {
+                    return modelName == "java.lang.String"
+                }
+                Object buildAll(String modelName, Project project) {
+                    return "It works!"
+                }
+            }
+        """.stripIndent()
+        buildFile << """
+            plugins {
+                id("java")
+            }
+            def registry = services.get(org.gradle.tooling.provider.model.ToolingModelBuilderRegistry)
+            registry.register(new my.MyModelBuilder())
+        """
+        file("gradle.properties") << """
+            $ENABLE_GRADLE_PROP=true
+        """.stripIndent()
+
+        expect:
+        usingToolingConnection(testDirectory) { connection ->
+            2.times {
+                def output = new ByteArrayOutputStream()
+                def model = connection.action(new SomeToolingBuildAction())
+                    .addJvmArguments(executer.jvmArgs)
+                    .forTasks("assemble")
+                    .setStandardOutput(output)
+                    .setStandardError(System.err)
+                    .run()
+                assert model == "It works!"
+                assert !output.toString().contains("Configuration cache is an incubating feature.")
```

Comment:
If we ever change this string, this test may silently stop checking anything. It might be better to, for example, add some logging to the build script and assert that this logging is present in both executions (that is, the build script is run in both cases).